### PR TITLE
Odis forno rate limiting fix

### PR DESCRIPTION
**Pull request by alecps**
### Description

Adds hotfix to combiner code to catch uncaught timeout errors caused by forno rate limiting.
Adds what is hopefully a fix for the forno error handling to the common package. This can be published and tested along with the next combiner release. 

My hypothesis for what is breaking the error handling is that previously we were accessing error.message in the catch block which sometimes fails since error is of type any. By switching to the try/catch syntax we are less likely to have subtle errors like this that go unnoticed. I should be able to test this theory by monitoring the logs during times when forno is causing timeouts. 

### Other changes

None

### Tested

Will be tested via monitoring in staging and alfajores before deploying to mainnet

### Related issues

- NA 

### Backwards compatibility

Yes

### Documentation

None

### CODE REVIEW

File: packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts [hunk 1]
```diff
@@ -47,7 +47,17 @@ export async function handleGetContactMatches(request: Request, response: Respon
       respondWithError(response, 400, WarningMessage.INVALID_INPUT, logger)
       return
     }
-    if (!(await authenticateUser(request, getContractKit(), logger))) {
+
```
Comment (eelanagaraj) [inline packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts:50]: is there any issue if no error is thrown? Or does this just then return an empty list of contact matches?

File: packages/phone-number-privacy/common/src/utils/authentication.ts [hunk 2]
```diff
@@ -108,33 +111,37 @@ export async function isVerified(
   contractKit: ContractKit,
   logger: Logger
 ): Promise<boolean> {
-  return retryAsyncWithBackOffAndTimeout(
```
Comment (codyborn) [inline packages/phone-number-privacy/common/src/utils/authentication.ts:111]: I believe you can also catch the error like:
`return retryAsyncWithBackOffAndTimeout(....).catch(error => { ... })`

**Review by codyborn — APPROVED** [review]
Looks good. Added small comment.

File: packages/phone-number-privacy/common/src/utils/authentication.ts (hunk 2 above)
Comment (alecps) [reply packages/phone-number-privacy/common/src/utils/authentication.ts:111]: Yea you should be able to. The reason I changed it to use the try/catch syntax is for better type checking on the error. When I switched to this syntax I uncovered what I think is the cause of the issue, so I feel like this syntax may be better for avoiding this type of bug.

File: packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts (hunk 1 above)
Comment (alecps) [reply packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts:50]: If I understand correctly you're asking if there's an issue with catching the error here even though the check was not succesful? We catch the error and assume the user is authenticated to provide better onboarding UX when there are availability issues.

Comment (eelanagaraj) [reply packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts:50]: I mean does it cause any confusing behavior later if we don't throw an error here? So from your response, is this correct: we continue as if everything is working as expected at this point, log the error, and at the very end we would return an empty list of matches to allow for a cleaner onboarding flow (as a temp solution) right? Just trying to understand if there's any issues with _not_ throwing an error here! Thanks 🙏

**Pull request merged**